shared/pkg/config/runtime_scanner.go

```diff
@@ -40,19 +49,131 @@ func setRegistryConfigDefaults() {
 func LoadRuntimeScannerRegistryConfig(imageID string) *Registry {
 	setRegistryConfigDefaults()
 
-	username, password, err := creds.ExtractCredentials(imageID)
-	if err != nil {
-		log.Warnf("Failed to extract credentials for imageID %v: %v", imageID, err)
+	var auth Auth
+
+	imagePullSecretPath := viper.GetString(ImagePullSecretPath)
+	if imagePullSecretPath != "" {
+		authConfig, err := GetAuthConfigFromPullSecrets(imagePullSecretPath, imageID)
+		if err != nil {
+			log.Fatalf("Failed to resolve auth config from pull secrets: %v", err)
+		}
+
+		if authConfig.Username != "" {
+			auth.Username = authConfig.Username
+		}
+
+		if authConfig.Password != "" {
+			auth.Password = authConfig.Password
+		}
+
+		if authConfig.RegistryToken != "" {
+			auth.Token = authConfig.RegistryToken
+		}
 	}
 
 	return &Registry{
 		SkipVerifyTLS: viper.GetBool(RegistrySkipVerifyTlS),
 		UseHTTP:       viper.GetBool(RegistryUseHTTP),
-		Auths: Auths{
-			{
-				Username: username,
-				Password: password,
+		Auths:         []Auth{auth},
+	}
+}
+
+func GetAuthConfigFromPullSecrets(ipsPath string, imageName string) (*authn.AuthConfig, error) {
```

Comment:
seems like `GetAuthConfigFromPullSecrets` and `NewKeyChainFromPath` should be in a util pkg, DYT?

Reply:
moving these into the existing shared/utils/creds pkg 👍